src/main/java/edu/nicholaidudakiwwarrick/advancedjava/services/DatabaseStockService.java

```diff
@@ -0,0 +1,117 @@
+            while(resultSet.next()) {
+                Timestamp timeStamp = resultSet.getTimestamp("time");
+                DateTime time = new DateTime(timeStamp);
+                BigDecimal price = resultSet.getBigDecimal("price");
+                stockQuotes.add(new StockQuote(symbol, price, time));
+            }
+        } catch (DatabaseConnectionException | SQLException exception) {
+            throw new StockServiceException(exception.getMessage(), exception);
+        }
+        if (stockQuotes.isEmpty()) {
+            throw new StockServiceException("There is no stock data for:" + symbol);
+        }
+        return stockQuotes.get(0);
+    }
+
+    /**
+     * Get a historical list of stock quotes for the provide symbol
+     *
+     * @param symbol the stock symbol to search for
+     * @param startDate   the date of the first stock quote
+     * @param endDate  the date of the last stock quote
+     * @return a list of StockQuote instances
+     * @throws   StockServiceException if using the service generates an exception.
+     * If this happens, trying the service may work, depending on the actual cause of the
+     * error.
+     */
+    @Override
+    public final List<StockQuote> getQuote(String symbol, DateTime startDate, DateTime endDate) throws StockServiceException {
+        List<StockQuote> stockQuotes = null;
+        try {
+            Connection connection = DatabaseUtils.getConnection();
+            Statement statement = connection.createStatement();
+            String queryString = "select id from stock_symbols where symbol = '" + symbol + "'";
+            ResultSet resultSet = statement.executeQuery(queryString);
+            resultSet.next();
+            queryString = "select * from quotes where symbol_id = '" + resultSet.getInt("id") + "'" +
+                    "AND time between '" + startDate.toString(StockQuote.DATE_PATTERN) + "' and '" +
+                    endDate.toString(StockQuote.DATE_PATTERN) + "'";
+
+            resultSet = statement.executeQuery(queryString);
+            stockQuotes = new ArrayList<>(resultSet.getFetchSize());
+            while(resultSet.next()) {
+                Timestamp timeStamp = resultSet.getTimestamp("time");
+                DateTime time = new DateTime(timeStamp);
+                BigDecimal price = resultSet.getBigDecimal("price");
+                stockQuotes.add(new StockQuote(symbol, price, time));
+            }
+        } catch (DatabaseConnectionException | SQLException exception) {
+            throw new StockServiceException(exception.getMessage(), exception);
+        }
+        if (stockQuotes.isEmpty()) {
+            throw new StockServiceException("There is no stock data for:" + symbol);
+        }
+        return stockQuotes;
+    }
+
+    @Override
+    public final List<StockQuote> getQuote(String symbol, DateTime startDate, DateTime endDate, IntervalEnum interval) {
+        return null;
```

Comment:
really try hard to never return null. 
in this case, you could just return an empty list. 
actually, in this case, you should just through unsupported operation exception.